src/examples/shared/useCalculateScroll.ts

```diff
@@ -0,0 +1,61 @@
+import { useLayoutEffect, useState } from "react"
+import analyticsEventSections from "./example-sections/constants"
+import { majorScale } from "evergreen-ui"
+
+const clamp = (value: number) => Math.max(0, value)
+
+const isBetween = (value: number, floor: number, ceil: number) =>
+  value >= floor && value <= ceil
+
+const useCalculateScroll = (
+  offset = majorScale(17)
+): { activeTitle: string } => {
+  const [activeTitle, setActiveTitle] = useState("")
+
+  useLayoutEffect(() => {
+    const listener = () => {
+      const scroll = window.pageYOffset
```

Comment:
Consider naming this something more specific like `scrollPosition` since `scroll` might be taken as an action/function name 👍